Assets/Scripts/Puzzle/Gate.cs

```diff
@@ -22,6 +23,7 @@ public class Gate : SignalReceiver {
 		player = playerObj.transform;
 		playerController = playerObj.GetComponent<PlayerController>();
 		camera = Camera.main.GetComponent<SnappingCamera>();
+		cameraTransitionDuration = camera.TransitionDuration;
 	}
 
 	public void Toggle() {
```

Comment:
Maybe have a guard variable for the coroutine to make sure that it can't be executed while it's already running?

Reply:
Could do, probably not needed though due to the player not being able to act during the coroutine. Can put one up just to be safe.